internal/server/server.go

```diff
@@ -317,6 +319,41 @@ func (s *Server) v1SourcesDataPut(w http.ResponseWriter, r *http.Request) {
 	JSONOK(w, resp, pretty(r))
 }
 
+func (s *Server) v1SourcesDataPatch(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	name, err := url.PathUnescape(r.PathValue("source"))
+	if err != nil {
+		ErrorString(w, http.StatusBadRequest, types.CodeInvalidParameter, err)
+		return
+	}
+
+	var patch types.SourcesPatchDataRequestV1
+	if err := newJSONDecoder(r.Body).Decode(&patch); err != nil {
+		writer.ErrorString(w, http.StatusBadRequest, types.CodeInvalidParameter, err)
+		return
+	}
+
+	previous, _, err := s.db.SourcesDataGetRaw(ctx, name, path.Join(r.PathValue("path"), "data.json"), s.auth(r))
```

Comment:
Should getting the data, applying the patch and saving it back be done in a transaction? Or ist that already automatted through `s.db`?

Reply:
Yup that's a good point. I'll try to move the patch application one layer down -- the database API does not expose transactions.

Reply:
Thanks for your thoughts. Let's roll with this. If we change it, the requirement would become laxer, which is always easier to introduce.